polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala

```diff
@@ -240,163 +240,297 @@ class PythonInterpreter private[python] (
+      |            sys.stdout = kernel.display
+      |            tracking_ns = TrackingNamespace(globals)
+      |            for stat, isImport in compiled:
+      |                if isImport: # don't track locals if the tree is an import.
+      |                    exec(stat, tracking_ns.globals)
+      |                else:
+      |                    exec(stat, tracking_ns)
+      |                    tracking_ns.globals.update(tracking_ns.data)
+      |
+      |            types = { x: type(y).__name__ for x,y in tracking_ns.data.items() }
+      |            return { 'globals': tracking_ns.globals, 'locals': tracking_ns.data, 'types': types }
+      |        except Exception as err:
+      |            import traceback
+      |
+      |            def handle_exception(typ, err_val, tb):
+      |                trace = ArrayList()
+      |                for frame in traceback.extract_tb(tb):
+      |                    trace.add(StackTraceElement(frame.filename.split("/")[-1], frame.name, frame.filename, frame.lineno))
+      |                result = {
+      |                  'stack_trace': trace,
+      |                  'message': getattr(err_val, 'message', str(err_val)),
+      |                  'class': typ.__name__,
+      |                  'err': err_val
+      |                }
+      |
+      |                # Lifted from IPython.core.ultratb
+      |                def get_chained_exception(exception_value):
+      |                    cause = getattr(exception_value, '__cause__', None)
+      |                    if cause:
+      |                        return cause
+      |                    if getattr(exception_value, '__suppress_context__', False):
+      |                        return None
+      |                    return getattr(exception_value, '__context__', None)
+      |
+      |                cause = get_chained_exception(err_val)
       |                if cause:
-      |                    return cause
-      |                if getattr(exception_value, '__suppress_context__', False):
-      |                    return None
-      |                return getattr(exception_value, '__context__', None)
-      |
-      |            cause = get_chained_exception(err_val)
-      |            if cause:
-      |                result['cause'] = handle_exception(type(cause), cause, cause.__traceback__)
+      |                    result['cause'] = handle_exception(type(cause), cause, cause.__traceback__)
       |
-      |            return result
+      |                return result
       |
-      |        return handle_exception(type(err), err, err.__traceback__)
+      |            return handle_exception(type(err), err, err.__traceback__)
+      |except Exception as e:
+      |    import sys, traceback
+      |    print("An error occurred while initializing the Python interpreter.", e, file=sys.stderr)
+      |    traceback.print_exc(file=sys.stderr)
+      |    sys.stderr.flush()
+      |    raise e
```

Comment:
This is a useful catch-all for figuring out why something went wrong in initialization, since we don't have all the error handling at this point.